Modules/+Drivers/ArduinoServo.m

```diff
@@ -0,0 +1,59 @@
+classdef ArduinoServo < Modules.Driver
+    %ARDUINOSERVO Connects with a hwserver Arduino and uses one pin to control a servo.
+    %
+    % Call with the 1) hostname of the host computer (singleton based on host), and 2) the (integer) pin.
+    
+    properties (Constant)
+        hwname = 'Arduino';
+    end
+    properties (SetAccess=immutable)
+        connection
+        pin
+    end
+    properties (GetObservable, SetObservable)
+        angle = Prefs.Double(NaN, 'min', 0, 'max', 180, 'set', 'set_angle', 'allow_nan', true);
+    end
+    methods(Static)
+        function obj = instance(host, pin)
+            mlock;
+            persistent Objects
+            if isempty(Objects)
+                Objects = Drivers.ArduinoServo.empty(1,0);
+            end
+            [~,resolvedIP] = resolvehost(host);
+            for i = 1:length(Objects)
+                if isvalid(Objects(i)) && isequal({resolvedIP, pin}, Objects(i).singleton_id)
+                    obj = Objects(i);
+                    return
+                end
+            end
+            obj = Drivers.ArduinoServo(resolvedIP, pin);
+            obj.singleton_id = {resolvedIP, pin};
+            Objects(end+1) = obj;
+        end
+    end
+    methods(Access=private)
+        function obj = ArduinoServo(host, pin)
+            obj.connection = hwserver(host);
+            obj.com('?');
+            obj.pin = pin;
+        end
+        function response = com(obj,funcname,varargin) %keep this
+            response = obj.connection.com(obj.hwname,funcname,varargin{:});
+        end
+    end
+    methods
+        function delete(obj)
+            delete(obj.connection)
+        end
+        function val = set_angle(obj,val,~)     % Locks to new angle (0 -> 180 standard), then unlocks.
+            obj.com(['s ' num2str(obj.pin) ' ' num2str(val)]);
+        end
```

Comment:
Should this be calling the lock and unlock methods, or is this command something that takes care of all of that on the servo?

Reply:
All from servo.